frontend/src/component/common/FavoriteIconButton/FavoriteIconButton.tsx

```diff
@@ -1,50 +1,58 @@
 import React, { VFC } from 'react';
-import { Dialogue } from 'component/common/Dialogue/Dialogue';
-import useFeatureApi from 'hooks/api/actions/useFeatureApi/useFeatureApi';
-import useToast from 'hooks/useToast';
-import { formatUnknownError } from 'utils/formatUnknownError';
-import { IconButton } from '@mui/material';
+import { IconButton, SxProps, Theme } from '@mui/material';
 import { ConditionallyRender } from '../ConditionallyRender/ConditionallyRender';
 import {
     Star as StarIcon,
     StarBorder as StarBorderIcon,
 } from '@mui/icons-material';
+import { TooltipResolver } from '../TooltipResolver/TooltipResolver';
 
 interface IFavoriteIconButtonProps {
     onClick: (event?: any) => void;
     isFavorite: boolean;
     size?: 'medium' | 'large';
+    sx?: SxProps<Theme>;
 }
 
 export const FavoriteIconButton: VFC<IFavoriteIconButtonProps> = ({
     onClick,
     isFavorite,
     size = 'large',
+    sx,
 }) => {
     return (
-        <IconButton size={size} data-loading sx={{ mr: 1 }} onClick={onClick}>
+        <IconButton
+            size={size}
+            data-loading
+            sx={{ mr: 1, ...sx }}
+            onClick={onClick}
+        >
             <ConditionallyRender
                 condition={isFavorite}
                 show={
-                    <StarIcon
-                        color="primary"
-                        sx={{
-                            fontSize: theme =>
-                                size === 'medium'
-                                    ? theme.spacing(2)
-                                    : theme.spacing(3),
-                        }}
-                    />
+                    <TooltipResolver title={'Remove from favorites'}>
```

Comment:
I'd extract those small components to have less whitespace on the left hand side of the code. It's difficult to read with prettier max line of 80 chars.